lib/unifex/code_generators/nif.ex

```diff
@@ -131,6 +131,16 @@ defmodule Unifex.CodeGenerators.NIF do
     ~g<UNIFEX_TERM #{name}(#{args_declarations})>
   end
 
+  defp generate_args_declarations(args, mode, ctx) do
+    Enum.flat_map(args, fn {name, type} ->
+      if Common.is_pointer_on_pointer_type(type, NIF, ctx) do
+        BaseType.generate_declaration(type, name, :default, NIF, ctx)
+      else
+        BaseType.generate_declaration(type, name, mode, NIF, ctx)
+      end
```

Comment:
Let's do that check inside `BaseType.generate_declaration`. Add a comment describing why we do that.

Reply:
Omiting `const` takes place in args declaration of `send` and `result` functions. Are you sure, that we want to omit `const` in every use of &BaseType.generate_declaration/5, not only in this 2 cases?

Reply:
Good point, we need a way to provide information whether that's needed or not. Maybe a separate `mode`?